index.js

```diff
@@ -140,6 +141,26 @@ if (semver.satisfies(sequelizeVersion, '4.x')) {
   }
 });
 
+// By default, Sequelize does not run ID in `WHERE primaryKey = ID` through
+// `$stringify` when said WHERE clause is generated via `model.update` or
+// `model.save`. ID is the exact object returned from the Postgres driver from
+// an earlier call to `model.findAll`, so it's reasonable for Sequelize to
+// assume ID could be handed back to the database directly--and it Postgres, it
+// always can. CockroachDB won't accept a stringified integer, though, so the
+// query will fail with "unsupported comparison: <int> = <string>".
+//
+// By hooking into every update query and setting `options.model` appropriately,
+// we enable the slightly-less-efficient code path that calls `$stringify` on
+// ID.
+QueryInterface.prototype.update = (function (oldUpdate) {
+  return function update(instance, tableName, values, identifier, options) {
+    // The value we need for `options.model` is thankfully passed to this
+    // function as the `instance` argument.
+    options = Object.assign({}, options, { model: instance });
```

Comment:
It's to avoid mutating `options`. It's the equivalent of

```
options = _.clone(options);
options.model = instance
```

but without introducing a dependency on underscore/lodash.